examples/slave/nuttx/xmc4800/EmulatedEEPROM.cc

```diff
@@ -0,0 +1,203 @@
+#include "EmulatedEEPROM.h"
+
+#include "kickcat/checksum/adler32.h"
+
+extern "C"
+{
+    #include <xmc4_flash.h>
+    #include <debug.h>
+}
+
+#include <cstring>
+
+
+namespace kickcat
+{
+    #define EEPROM_CTRL_STAT_ECAT_WR_EN  (1 << 0)
+    #define EEPROM_CTRL_STAT_EMULATION   (1 << 5)
+    #define EEPROM_CTRL_STAT_READ_BYTES  (1 << 6)
+    #define EEPROM_CTRL_STAT_ALGO_SEL    (1 << 7)
+    #define EEPROM_CTRL_STAT_COMMAND     (7 << 8)
+    #define EEPROM_CTRL_STAT_CRC         (1 << 11)
+    #define EEPROM_CTRL_STAT_LOADING     (1 << 12)
+    #define EEPROM_CTRL_STAT_ERROR_CMD   (1 << 13)
+    #define EEPROM_CTRL_STAT_ERROR_WR_EN (1 << 14)
+    #define EEPROM_CTRL_STAT_BUSY        (1 << 15)
+
+    #define EEPROM_READ_BYTES 8
+
+    // XMC specific
+    constexpr uint32_t ECAT_EEP_CONF = ECAT0_BASE_ADDRESS + reg::EEPROM_CONFIG;
```

Comment:
Better to have a template function that return a pointer on the desired memory with the desired pointer:
```cpp
template<typename T>
T* address(uint32_t offset)
{   
   void* address =  ECAT0_BASE_ADDRESS + offset);
    return static_cast<T*>(address);
}
```